src/Sentry/Internal/MainExceptionProcessor.cs

```diff
@@ -47,9 +51,33 @@ public void Process(Exception exception, SentryEvent sentryEvent)
                     continue;
                 }
 
-                foreach (var key in sentryException.Data.Keys)
+                foreach (var keyValue in sentryException.Data)
                 {
-                    sentryEvent.SetExtra($"Exception[{i}][{key}]", sentryException.Data[key]);
+                    if (keyValue.Key.StartsWith("sentry:", StringComparison.OrdinalIgnoreCase) &&
+                        keyValue.Value != null)
+                    {
+                        if (keyValue.Key.StartsWith(ExceptionDataTagKey, StringComparison.OrdinalIgnoreCase) &&
+                            keyValue.Value is string tagValue)
+                        {
+                            //Set the key after the ExceptionDataTagKey string.
+                            sentryEvent.SetTag(keyValue.Key.Substring(ExceptionDataTagKey.Length), tagValue);
+                        }
+                        else if (keyValue.Key.StartsWith(ExceptionDataContextKey, StringComparison.OrdinalIgnoreCase))
+                        {
+                            //Set the key after the ExceptionDataTagKey string.
+                            _ = sentryEvent.Contexts.AddOrUpdate(keyValue.Key.Substring(ExceptionDataContextKey.Length),
```

Comment:
Curious why using AddOrUpdate here instead of plain: `[..] =` ?

The code still doesn't account for the key being an exact match to the prefix. It would use an empty string as key `""`. We should make sure the string is larger than the key at least.

Reply:
Not sure why, maybe I saw it at some old Sample/Doc.